api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -102,6 +102,15 @@ default String name() {
    */
   String location();
 
+  /**
+   * Returns the location of the current metadata file.
+   *
+   * @return this table's metadata file location
+   */
+  default String metadataFileLocation() {
+    return null;
```

Comment:
For backward compatibility.

Reply:
Shouldn't it be not implemented exception? Or is it ok for this to return null sometimes?

Reply:
I was thinking about this one too. Probably, an exception would be more descriptive? I am going to use it in the serializable table, where it will be optional. So null there is no problem but I am not sure about general cases.

Reply:
Yeah my basic thought here was that you shouldn't be calling this if you are ok with it being null, so we should throw an error

Reply:
Reverted this.